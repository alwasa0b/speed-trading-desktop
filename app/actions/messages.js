import {
  UPDATE_POSITIONS,
  UPDATE_ORDERS,
  UPDATE_PRICE,
  PRICE_UPDATED,
  POSITIONS_UPDATED,
  ORDERS_UPDATED
} from "../constants/messages";

const { ipcRenderer } = require("electron");

export const update_positions = () => async (
  dispatch,
  getState,
  ipc = ipcRenderer
) => {
  await ipc.removeAllListeners(POSITIONS_UPDATED);
  await ipc.send(UPDATE_POSITIONS);

  ipc.on(POSITIONS_UPDATED, async (event, data) => {
    dispatch({ type: POSITIONS_UPDATED, data });
  });
};

export const update_orders = () => async (
  dispatch,
  getState,
  ipc = ipcRenderer
) => {
  await ipc.removeAllListeners(ORDERS_UPDATED);
  await ipc.send(UPDATE_ORDERS);

  ipc.on(ORDERS_UPDATED, async (event, data) => {
    dispatch({ type: ORDERS_UPDATED, data });
  });
};

export const update_price = ({ symbol }) => async (
  dispatch,
  getState,
  ipc = ipcRenderer
) => {
  await ipc.removeAllListeners(PRICE_UPDATED);
  await ipc.send(UPDATE_PRICE, { symbol });

  ipc.on(PRICE_UPDATED, async (event, data) => {
    dispatch({ type: PRICE_UPDATED, data });
  });
};
