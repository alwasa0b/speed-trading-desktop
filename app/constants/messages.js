export const UPDATE_PRICE = "UPDATE_PRICE";
export const UPDATE_POSITIONS = "UPDATE_POSITIONS";
export const UPDATE_ORDERS = "UPDATE_ORDERS";
export const PRICE_UPDATED = "PRICE_UPDATED";
export const POSITIONS_UPDATED = "POSITIONS_UPDATED";
export const ORDERS_UPDATED = "ORDERS_UPDATED";

export const START_WORKER = "START_WORKER";
export const WORKER_STARTED = "WORKER_STARTED";
export const STOP_WORKER = "STOP_WORKER";
export const WORKER_STOPPED = "WORKER_STOPPED";
export const PAUSE_WORKER = "PAUSE_WORKER";
export const WORKER_PAUSED = "WORKER_PAUSED";
export const RESUME_WORKER = "RESUME_WORKER";
export const WORKER_RESUMED = "WORKER_RESUMED";
export const UPDATE_WORKER = "UPDATE_WORKER";
export const WORKER_UPDATED = "WORKER_UPDATED";
export const PANIC = "PANIC";
export const TOGGLE_LOCKED = "TOGGLE_LOCKED";

export const PROGRESS_UPDATE = "PROGRESS_UPDATE";
export const START_LOGGING = "START_LOGGING";

export const INFO = "INFO";
export const ERROR = "ERROR";
export const WARN = "WARN";

export const ADD_SYMBOL = "ADD_SYMBOL";
export const SYMBOL_UPDATE = "SYMBOL_UPDATE";
export const REMOVE_SYMBOL = "REMOVE_SYMBOL";

export const TICKER_UPDATED = "TICKER_UPDATED";
