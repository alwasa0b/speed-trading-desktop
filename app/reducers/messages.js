import {
    POSITIONS_UPDATED,
    ORDERS_UPDATED,
    PRICE_UPDATED
  } from "../constants/messages";


export default (state = {}, action) => {
    switch (action.type) {
      case POSITIONS_UPDATED:
        return { ...state, positions: action.data };
      case ORDERS_UPDATED:
        return { ...state, orders: action.data };
      case PRICE_UPDATED:
        return { ...state, price: action.data };
      default:
        return state;
    }
  }
  