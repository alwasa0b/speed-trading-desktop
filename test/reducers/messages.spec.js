import messages from "../../app/reducers/messages";
import {
  POSITIONS_UPDATED,
  ORDERS_UPDATED,
  PRICE_UPDATED
} from "../../app/constants/messages";

describe("reducers", () => {
  describe("messages", () => {
    it("should handle initial state", () => {
      expect(messages(undefined, {})).toMatchSnapshot();
    });

    it("should handle POSITIONS_UPDATED", () => {
      expect(
        messages(undefined, {
          type: POSITIONS_UPDATED,
          data: [{ symbol: "symbol" }]
        })
      ).toMatchSnapshot();
    });

    it("should handle ORDERS_UPDATED", () => {
      expect(
        messages(undefined, {
          type: ORDERS_UPDATED,
          data: [{ symbol: "symbol" }]
        })
      ).toMatchSnapshot();
    });

    it("should handle PRICE_UPDATED", () => {
      expect(
        messages(undefined, {
          type: PRICE_UPDATED,
          data: 10
        })
      ).toMatchSnapshot();
    });
  });
});
