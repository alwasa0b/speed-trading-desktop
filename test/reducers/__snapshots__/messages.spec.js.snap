// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`reducers messages should handle ORDERS_UPDATED 1`] = `
Object {
  "orders": Array [
    Object {
      "symbol": "symbol",
    },
  ],
}
`;

exports[`reducers messages should handle POSITIONS_UPDATED 1`] = `
Object {
  "positions": Array [
    Object {
      "symbol": "symbol",
    },
  ],
}
`;

exports[`reducers messages should handle PRICE_UPDATED 1`] = `
Object {
  "price": 10,
}
`;

exports[`reducers messages should handle initial state 1`] = `Object {}`;
